### ISSUE TITLE
No default ruby?

### ISSUE THREAD

**Issue by karl-petter** [rbenv/rbenv#926]
I'm quite new to rbenv and just stumbled into a problem that I do not know if it is a bug or expected behaviour.

I installed rbenv via brew and run rbenv init. After closing my shell and opening a new one I execute `ruby --version` to see which ruby it will use. It generates the following error which is quite confusing:
`/usr/local/Cellar/rbenv/1.0.0/libexec/rbenv-exec: line 47: /Users/kalle/.rbenv/shims/ruby: Argument list too long`

I realised this happens due to that I do not have any ruby version set(since I just installed it from scratch). Is this expected behaviour?

Some information that might be helpful:

```
> rbenv local
rbenv: no local version configured for this directory

> rbenv global
system

> rbenv versions
* system (set by /Users/kalle/.rbenv/version)
  2.1.3
  2.2.5

> rbenv which ruby
/Users/kalle/.rbenv/shims/ruby

> /usr/bin/ruby --version
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
```

**Comment by mislav** [comment]
That is **very** strange. It looks like it may be recursion: i.e. one command invoking itself over and over again.

1. What is the output of `which -a rbenv`
2. What is the output of `which -a ruby`
3. Did you install any rbenv plugins?
4. What is your PATH?

**Comment by karl-petter** [comment]
Yes it is some kind of recursion. I see in my shell header that it is calling rbenv again and again.

```
> which -a rbenv
/usr/local/bin/rbenv

> which -a ruby
/Users/kalle/.rbenv/shims/ruby
/usr/bin/ruby

> echo $PATH
/Users/kalle/.rbenv/shims:~/.rbenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```
which all looks good to me.

Regarding plugins, none that I'm aware of. Can I somehow check that?

And just to point out; if I create a ~/.rbenv/version file all works fine. The brew installation was not supposed to have created one?

**Comment by mislav** [comment]
Oh wow. You helped me discover an edge-case bug in rbenv. You had `shims` twice in PATH:

```
/Users/kalle/.rbenv/shims:~/.rbenv/shims
```

First, you only need it once. Second, it seems like rbenv doesn't handle well the format with the tilde `~/.rbenv/shims`. In general, you should not have tilde `~` in PATH, for compatibility with most programs.

I can change rbenv to handle this better, but in the meantime, find where you add `~/.rbenv/shims` to PATH and eliminate it. The `/Users/kalle/.rbenv/shims` entry is fine and likely added by the `eval "$(rbenv init -)"` call in your shell initialization file if you configured rbenv correctly.

**Comment by karl-petter** [comment]
Aah, wow. But great to be of help! I'll fix that on Monday.

**Comment by mislav** [comment]
If you have further problems after editing your PATH, just let me know!

**Comment by karl-petter** [comment]
Sorry but forgot about this. Removed the PATH addition and now all works just fine. Thanks!

**Comment by mislav** [comment]
Fixes in d0779fc8fb90fb7268d0ee858086e92bcd316787
